roles/reverseproxy/templates/nginx.conf.j2

```diff
@@ -0,0 +1,38 @@
+server_names_hash_bucket_size 128;
+resolver 8.8.8.8 ipv6=off;
+
+{% for server,urls in reverseproxy_sites.items() %}
+server {
+        listen 80;
+        listen 443 ssl;
+        server_name {{server}};
+        ssl_certificate /etc/ssl/reverseproxy.cert;
+        ssl_certificate_key /etc/ssl/reverseproxy.key;
+
+        # Tune some buffer settings
+        proxy_buffer_size 16k;
+        proxy_busy_buffers_size 24k;
+        proxy_buffers 64 4k;
+
+        # Set the hostname as a variable, this forces nginx to do a dns lookup
+        # otherwise it only looks it up when it starts and never refreshes it
+        set $server_host "{{server}}";
```

Comment:
Why is this needed?

Reply:
Because the resources are mostly hosted on CDNs, and they typically have short TTLs on their DNS entries/might rotate an IP out of service during the contest. If nginx only looks it up on start, then if the dns entry changes/the original IP stops working, nginx will never look it up again.